src/views/search/search.jsx

```diff
@@ -47,6 +51,20 @@ class Search extends React.Component {
         }
         term = decodeURIComponent(term.split('+').join(' '));
         this.props.dispatch(navigationActions.setSearchTerm(term));
+        
+        let mode = '';
+        const m = query.lastIndexOf('mode=');
+        if (m !== -1) {
+            mode = query.substring(m + 5, query.length).toLowerCase();
+        }
+        while (mode.indexOf('/') > -1) {
+            mode = mode.substring(0, term.indexOf('/'));
+        }
+        while (term.indexOf('&') > -1) {
+            mode = mode.substring(0, term.indexOf('&'));
+        }
+        mode = decodeURIComponent(mode.split('+').join(' '));
+        this.props.dispatch(navigationActions.setMode(mode));
```

Comment:
Here, instead of dispatching, just use `this.setState({mode: mode})` to set the `mode` state.

Reply:
I got a Travis error when I did this https://travis-ci.org/LLK/scratch-www/builds/419141652#L761

Reply:
Oh that's right — none of this needs to happen in `componentDidMount`, it can all happen in the constructor.  And if you do it in the constructor, then you don't need to use `setState`, you can just initialize the state with this value.